Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -82,26 +62,119 @@ public States States
         /// <summary>
         /// The state logic for comparing state
         /// </summary>
-        public InteractableStates StateManager;
+        public InteractableStates StateManager { get; protected set; }
 
         /// <summary>
         /// Which action is this interactable listening for
         /// </summary>
-        public MixedRealityInputAction InputAction;
+        public MixedRealityInputAction InputAction { get; set; }
 
-        // the id of the selected inputAction, for serialization
+        /// <summary>
+        /// The id of the selected inputAction, for serialization
+        /// </summary>
         [HideInInspector]
-        public int InputActionId;
+        [SerializeField]
+        private int InputActionId = -1;
 
+        [FormerlySerializedAs("IsGlobal")]
+        [SerializeField]
+        protected bool isGlobal = false;
         /// <summary>
         /// Is the interactable listening to global events (input only)
         /// </summary>
-        public bool IsGlobal = false;
+        public bool IsGlobal
+        {
+            get { return isGlobal; }
+            set
+            {
+                if (isGlobal != value)
+                {
+                    isGlobal = value;
+
+                    // If we are active, then change global speech registeration. 
+                    // Register handle if we do not require focus, unregister otherwise
+                    if (gameObject.activeInHierarchy)
+                    {
+                        RegisterHandler<IMixedRealityInputHandler>(isGlobal);
+                    }
+                }
+            }
+        }
 
+        [FormerlySerializedAs("Dimensions")]
+        [SerializeField]
+        protected int dimensions = 1;
         /// <summary>
         /// A way of adding more layers of states for controls like toggles
         /// </summary>
-        public int Dimensions = 1;
+        public int Dimensions
+        {
+            get { return dimensions; }
+            set
+            {
+                // Value cannot be negative or zero
+                if (value > 0 && dimensions != value)
```

Comment:
>if (value > 0 && dimensions != value) [](start = 16, length = 37)

consider putting an assert statement to call out when invalid values are passed in